scripts/base/protocols/redis/spicy-decls.zeek

```diff
@@ -0,0 +1,97 @@
+##! Events and records generated by the Redis analyzer.
+
+module Redis;
+
+export {
+	## The Redis SET command.
+	type SetCommand: record {
+		## The key the SET command is setting.
+		key: string &log;
+		## The value the SET command is setting key to.
+		value: string &log;
+		## If NX is set -- only set the key if it does not exist.
+		nx: bool;
+		## If XX is set -- only set the key if it already exists.
+		xx: bool;
+		## If GET is set -- return the old string stored at key.
+		get: bool;
+		## EX option -- set the specified expire time, in seconds.
+		ex: count &optional;
+		## PX option -- set the specified expire time, in milliseconds.
+		px: count &optional;
+		## EXAT option-- set the specified Unix time at which the key will
+		## expire, in seconds.
+		exat: count &optional;
+		## PXAT option -- set the specified Unix time at which the key will
+		## expire, in milliseconds.
+		pxat: count &optional;
+		## If KEEPTTL is set -- retain the time to live associated with the key.
+		keep_ttl: bool;
+	};
+
+	## The Redis AUTH command.
+	type AuthCommand: record {
+		## The username getting authenticated.
+		username: string &optional;
+		## The password authenticated with.
+		password: string;
+	};
+
+	## A generic Redis command from the client.
+	type Command: record {
+		## The raw command, exactly as parsed
+		raw: vector of string;
+		## The first element of the command. Some commands are two strings, meaning
+		## this is inaccurate for those cases.
+		command: string &log;
+		## The key, if this command is known to have a key
+		key: string &log &optional;
+		## The value, if this command is known to have a value
+		value: string &log &optional;
+		## The command in an enum if it was known
+		known: KnownCommand &optional;
+	};
+
+	## Generic server data returned from the server.
+	type ServerData: record {
+		## Was this an error?
+		err: bool &log;
```

Comment:
For discussion, maybe? IMO at least in the log this should be inverted to a `success` field (possibly within `Info` rather than in `Reply` so that the command/reply pair indicates success or failure)

If a user wants to query for successful responses using this log, their query is going to be `reply.err=F`, which doubly negative and makes me shudder (might be a me problem, too). It's also nicer to see `"success": true` in a log, rather than `"err": false`.

For the case of "no clue what happened with the command", it seems better to have "reply.success" unset, rather than "reply.err" unset.

There's other logs that have an artificial `success` field (e.g. mysql). `err` or `error` seem less common.

Reply:
Yeah I think success seems clearer, it's mostly just easier to think about an error field getting set when making the parser, but the user would find the opposite more intuitive, so it's probably a good swap